packages/visual-editor/src/utils/resolveTranslatableString.tsx

```diff
@@ -5,24 +5,34 @@ import {
   TranslatableString,
 } from "@yext/visual-editor";
 import React from "react";
+import { resolveEmbeddedFieldsRecursively } from "./resolveYextEntityField";
 
 /**
  * Converts a type TranslatableString to a string
  * @param translatableString
  * @param locale
+ * @param document
  */
 export const resolveTranslatableString = (
   translatableString: TranslatableString = "",
-  locale: string = "en"
+  locale: string = "en",
+  document?: any
 ): string => {
+  let resolvedString;
   if (typeof translatableString === "object") {
     if (locale in translatableString) {
-      return translatableString[locale];
+      resolvedString = translatableString[locale];
+    } else {
+      return "";
     }
-    return "";
+  } else {
+    resolvedString = translatableString;
   }
 
-  return translatableString;
+  // If the document is provided, resolve any embedded fields in the string.
```

Comment:
Is there a case where we don't just want to always do this and make the document required?